app/src/main/java/ch/epfl/culturequest/ui/notifications/NotificationsRecycleViewAdapter.java

```diff
@@ -42,11 +47,42 @@ public NotificationsRecycleViewAdapter.NotificationViewHolder onCreateViewHolder
     @Override
     public void onBindViewHolder(@NonNull NotificationsRecycleViewAdapter.NotificationViewHolder holder, int position) {
         holder.getNotificationText().setText(notificationTexts.get(position).getText());
-        setIconNotification(holder.getNotificationIcon(), notificationTexts.get(position).getChannelId());
+        setIconNotification(holder.getNotificationIcon(), notificationTexts.get(position).getChannelId(), notificationTexts.get(position).getSenderId());
+
         holder.getDeleteButton().setOnClickListener(view -> {
             Database.deleteNotification(Profile.getActiveProfile().getUid(), notificationTexts.get(position));
             notificationTexts.remove(position);
             notifyItemRemoved(position);
+            notifyItemRangeChanged(0, getItemCount());
+        });
+
+        holder.itemView.setOnClickListener(view -> {
+            System.out.println(notificationTexts.get(position).getChannelId());
+            switch (notificationTexts.get(position).getChannelId()) {
+                case ScanNotification.CHANNEL_ID:
```

Comment:
Why don't you use "notificationTexts.get(position).buildnotification().getPendingIntent().send();" instead of coding it again here ?

Reply:
I cannot find getPendingIntent, are you sure the Notification class implement this method?